packages/streamline/src/components/cards/card-carousel/card-carousel.tsx

```diff
@@ -93,7 +108,7 @@ export const CardCarousel = (props: CardCarouselProps) => {
           {dismissAction ? (
             <ButtonIcon
               testID={`${testID}-close-button`}
-              isLoading={isDismissing || isLoading}
+              isLoading={isBusy}
```

Comment:
Shouldn't it be loading only if `isDismissing`?

Reply:
I thought it might be interesting to show loading in all situations. But I can remove it.

Reply:
Ok then you should update the above condition `dismissAction ? [...] : null` because if there is no dismissAction and the onPress is ongoing, you won't display the loader then.

Reply:
Before your answered I pushed a new version of the code, I'm waiting your opinion on it : 

  const [handlePress, isResolving] = usePress({ onPress });
  const [handleOnDismiss, isDismissing] = usePress({ onPress: dismissAction?.onDismiss });
  const isBusy = isResolving || isLoading;

        <Box>
          {dismissAction || isBusy ? (
            <ButtonIcon
              testID={`${testID}-close-button`}
              isLoading={isDismissing || isBusy}
              iconName="Cross"
              accessibilityLabel={dismissAction?.accessibilityLabel ?? ''}
              appearance="light"
              onPress={handleOnDismiss}
              withContainer
            />
          ) : null}
        </Box>

Reply:
Conflicted, it looks like it will work fine and for readability, I would rather render a `top-right-content.tsx` where we could handle "readably" the condition so that everybody understands and don't break it if we work on it again later on.

And in this component, when loading, I would render a `spinner` directly and not a fake `button-icon`. 

WDYT?

Reply:
I agree with the first comment, i'll add an external component : `top-right-content.tsx`

But for your second idea, I already tried to add a `spinner` component, the render is different. With the `button-icon` (spinner) there's a circle grey background behind. That's not the case for the classic `spinner`. 
We can always add manually the background ourself, but if one day we will change the design of the` button-icon`, the coherence will be broken.

It's up to you.

<img width="42" alt="Capture d’écran 2023-09-28 à 10 13 12" src="https://github.com/GetLuko/streamline/assets/56963543/10914395-81ff-45e8-8d0f-164fe0679038">

Reply:
Let's do the way you just described

Reply:
ButtonIcon keeped